File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    if (s.length == 0) return true;
    const stack = [];
    for (let i = 0; i < s.length; i++) {
        if (s[i] == ')' || s[i] == ']' || s[i] == '}') {
            if (stack.length == 0) {
                return false;
            }
            if (s[i] == ")" && stack[stack.length - 1] == "(") {
                stack.pop();
            }
            else if (s[i] == "]" && stack[stack.length - 1] == "[") {
                stack.pop();
            }
            else if (s[i] == "}" && stack[stack.length - 1] == "{") {
                stack.pop();
            }
            else {
                stack.push(s[i]);
            }
        }
        else {
            stack.push(s[i]);
        }
    }

    if (stack.length == 0) return true;

    return false;




};

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;

        int n = s.size();

        for(int i=0;i<n;i++){
            if(st.size()==0 && (s[i]==')' || s[i]=='}' || s[i]==']')){
                return false;
            }
            if(s[i]=='(' || s[i]=='[' || s[i]=='{'){
                st.push(s[i]);
            }
            else{
                if(st.top()=='(' && s[i]==')'){
                    st.pop();

                }
                else if(st.top()=='[' && s[i]==']'){
                    st.pop();

                }
                else if(st.top()=='{' && s[i]=='}'){
                    st.pop();

                }
                else{
                    st.push(s[i]);
                }
            }
        }
        return (st.empty()?true:false);

        
    }
};